test/post_processor/test_dual_from_empirical_frequencies.py

```diff
@@ -10,33 +10,39 @@
 
 """Tests for the `dual_from_empirical_frequencies` function."""
 
+from copy import deepcopy
 from unittest import TestCase
 
 import numpy as np
-from numpy.random import default_rng
 from povm_toolbox.library import (
+    ClassicalShadows,
     RandomizedProjectiveMeasurements,
 )
 from povm_toolbox.post_processor import (
     POVMPostProcessor,
     dual_from_empirical_frequencies,
 )
+from povm_toolbox.quantum_info import MultiQubitPOVM
 from povm_toolbox.sampler import POVMSampler
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
 from qiskit.circuit.random import random_circuit
 from qiskit.primitives import StatevectorSampler
 from qiskit.quantum_info import (
     DensityMatrix,
+    Operator,
     SparsePauliOp,
 )
 
 
 class TestDualFromEmpiricalFrequencies(TestCase):
     """Test that we can construct optimal dual of a POVM from empirical frequencies."""
 
-    def test_empirical_dual(self):
-        """Test that the method constructs a valid dual."""
-        qc = random_circuit(2, 1, measure=False, seed=12)
-        rng = default_rng(96568)
+    RNG_SEED = 30
+
+    def setUp(self) -> None:
+        super().setUp()
+        qc = random_circuit(2, 1, measure=False, seed=self.RNG_SEED)
```

Comment:
`random_circuit` is not guaranteed to return the same circuit (even with a fixed seed) across major versions of Qiskit (e.g. a 1.X to 2.0 version bump).

Could you replace this with a hard-coded circuit? Either in-line or some QASM file which you load?
The reason that is important, is because the expectation values we check against below are hard-coded. If these were computed separately for the circuit we are using, we would not have to rely on the specific circuit being used as much.

Reply:
What you could do, is use the `random_circuit` that you get today, save it to a QASM file, and use that.
In that case, if you could include a comment somewhere which explains how you obtain the stored circuit, that would be ideal

Reply:
Okay good to know 👍 
Will do this